rasa/core/migrator.py

```diff
@@ -0,0 +1,214 @@
+
+        Returns:
+            Conversation IDs that are both requested and contained in the tracker
+            store. If no conversation IDs are requested, all conversation IDs in the
+            tracker store are returned.
+
+        """
+        conversation_ids_in_tracker_store = set(self.tracker_store.keys())
+
+        if not conversation_ids_in_tracker_store:
+            raise NoConversationsInTrackerStore(
+                f"Could not find any conversations in connected tracker store. "
+                f"Please validate your `endpoints.yml` and make sure the defined "
+                f"tracker store exists. Exiting."
+            )
+
+        if not self.requested_conversation_ids:
+            return conversation_ids_in_tracker_store
+
+        missing_ids_in_tracker_store = (
+            set(self.requested_conversation_ids) - conversation_ids_in_tracker_store
+        )
+
+        if missing_ids_in_tracker_store:
+            cli_utils.print_warning(
+                f"Could not find the following requested "
+                f"conversation IDs in connected tracker store: "
+                f"{', '.join(sorted(missing_ids_in_tracker_store))}"
+            )
+
+        conversation_ids_to_process = conversation_ids_in_tracker_store & set(
+            self.requested_conversation_ids
+        )
+
+        if not conversation_ids_to_process:
+            raise NoEventsToMigrate(
+                "Could not find an overlap between the requested "
+                "conversation IDs and those found in the tracker store. Exiting."
+            )
+
+        return conversation_ids_to_process
+
+    def _fetch_events_within_time_range(self) -> List[Dict[Text, Any]]:
+        """Fetch all events for `conversation_ids` within the supplied time range.
+
+        Returns:
+            List of serialized events with added `sender_id` field.
+
+        """
+        conversation_ids_to_process = self.get_conversation_ids_to_process()
+
+        cli_utils.print_info(
+            f"Fetching events for {len(conversation_ids_to_process)} "
+            f"conversation IDs:"
+        )
+
+        events = []
+
+        for conversation_id in tqdm(conversation_ids_to_process, "conversation IDs"):
+            tracker = self.tracker_store.retrieve(conversation_id)
+            if not tracker:
+                logger.info(
+                    f"Could not retrieve tracker for conversation ID "
+                    f"'{conversation_id}'. Skipping."
+                )
+                continue
+
+            _events = tracker.current_state(EventVerbosity.ALL)["events"]
+
+            if not _events:
+                logger.info(
+                    f"No events to migrate for conversation ID '{conversation_id}'."
+                )
+                continue
+
+            # the conversation IDs are needed in the event publishing
+            for event in _events:
+                event["sender_id"] = conversation_id
+                events.append(event)
```

Comment:
could also be in a function `_get_events_for_conversation_id` to  improve readability